src/components/frame/Extensions/pages/ColonyDetailsPage/ColonyDetailsPage.tsx

```diff
@@ -1,18 +1,106 @@
 import React, { FC } from 'react';
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
 
 import Navigation from '~v5/common/Navigation';
 import TwoColumns from '~v5/frame/TwoColumns';
 import Spinner from '~v5/shared/Spinner';
+import styles from './ColonyDetailsPage.module.css';
+import { useColonyContext, useMobile } from '~hooks';
+import Avatar from '~v5/shared/Avatar';
+import Icon from '~shared/Icon';
+import Button from '~v5/shared/Button';
+import ObjectiveBox from '~v5/common/ObjectiveBox';
 
 const displayName = 'frame.Extensions.pages.ColonyDetailsPage';
 
-const ColonyDetailsPage: FC = () => (
-  <Spinner loadingText={{ id: 'loading.colonyDetailsPage' }}>
-    <TwoColumns aside={<Navigation pageName="extensions" />}>
-      Colony Details page
-    </TwoColumns>
-  </Spinner>
-);
+const ColonyDetailsPage: FC = () => {
+  const { formatMessage } = useIntl();
+  const isMobile = useMobile();
+  const { colony } = useColonyContext();
+  const { name, metadata } = colony || {};
+  const { avatar, thumbnail } = metadata || {};
+
+  const socialLinks = [];
+
+  return (
+    <Spinner loadingText={{ id: 'loading.colonyDetailsPage' }}>
+      <TwoColumns aside={<Navigation pageName="extensions" />}>
+        <div className={clsx('pt-[4.375rem] px-6 pb-6 mt-10', styles.box)}>
+          <div className="absolute left-6 -top-11 rounded-full border-4 border-gray-100 flex">
+            <Avatar size="xm" avatar={avatar || thumbnail || ''} />
+          </div>
+          <h2 className="heading-2 mb-1">{name}</h2>
+          {/* @TODO: Add description when it will be in API */}
+          <p className="text-md text-gray-600 mb-5">
+            SatelliteDAO is a decentralised community that focuses on supporting
+            the launch satellites and space exploration initiatives.
+          </p>
+          <div className="mb-6">
+            {/* @TODO: Add functionality to add and display social links */}
+            {socialLinks.length ? (
+              <div />
+            ) : (
+              <button
+                type="button"
+                className="flex items-center text-md transition-colors hover:text-blue-400"
```

Comment:
Maybe would be good to use TextButton here (a new variant with an optional icon and other styles like font size )?

Reply:
Added as mode